pkg/exporter/syslogexporter/config.go

```diff
@@ -52,23 +52,28 @@ type Config struct {
 
 // Validate the configuration for errors. This is required by component.Config.
 func (cfg *Config) Validate() error {
+	invalidFields := []string{}
 	if cfg.Port < 1 || cfg.Port > 65525 {
-		return unsupportedPort
+		invalidFields = append(invalidFields, unsupportedPort)
 	}
 
 	if !net.IsFQDN(cfg.Endpoint) && !net.IsIPAddr(cfg.Endpoint) && cfg.Endpoint != "localhost" {
-		return invalidEndpoint
+		invalidFields = append(invalidFields, invalidEndpoint)
 	}
 
 	if strings.ToLower(cfg.Protocol) != "tcp" && strings.ToLower(cfg.Protocol) != "udp" {
-		return unsupportedProtocol
+		invalidFields = append(invalidFields, unsupportedProtocol)
 	}
 
 	switch cfg.Format {
 	case formatRFC3164Str:
 	case formatRFC5424Str:
 	default:
-		return unsupportedFormat
+		invalidFields = append(invalidFields, unsupportedFormat)
+	}
+
+	if len(invalidFields) > 0 {
+		return fmt.Errorf("%s", strings.Join(invalidFields, "; "))
```

Comment:
There is `errors.Join` function which could do the trick, but for some reason it is not used in otelcol code at all :confused: Anyway after few days of thinking about it I think we should use it

Reply:
I know why. They are using `multierr.Combine` :D

Reply:
Please use `invalidFields := []error` and then `return multierr.Combine(invalidFields...)`

Reply:
I made this change